app/src/main/java/com/example/firstandroid/Presentation/UpdateBathroom/UpdateBathroomUI.kt

```diff
@@ -10,72 +10,87 @@ import androidx.compose.material3.Checkbox
 import androidx.compose.material3.Text
 import androidx.compose.material3.TextField
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
 import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.unit.dp
 import kotlinx.coroutines.launch
-import java.math.BigDecimal
 
 @Composable
 fun UpdateBathroomDialog(
     onDismissRequest: () -> Unit,
     viewModel: UpdateBathroomViewModel,
     originalBathroomInfo: UpdateBathroomData
 ) {
+
     val state = viewModel.state
     val scope = rememberCoroutineScope()
-
```

Comment:
On a future PR you can move the state from the Composable to the Viewmodel.

The Viewmodel will hold all these values (title, location, capacity, free, cost, hours, etc). Then when you call viewModel.updateTitle for example, the Viewmodel will update the title data, and the view will observe that change.